networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -0,0 +1,306 @@
+from networkx.algorithms.isomorphism.vf2pp_helpers.candidates import _find_candidates
+from networkx.algorithms.isomorphism.vf2pp_helpers.feasibility import _feasibility
+from networkx.algorithms.isomorphism.vf2pp_helpers.node_ordering import _matching_order
+from networkx.algorithms.isomorphism.vf2pp_helpers.state import (
+    _restore_Tinout,
+    _update_Tinout,
+)
+
+__all__ = ["vf2pp_mapping", "vf2pp_is_isomorphic", "vf2pp_all_mappings"]
+
+_GraphParameters = collections.namedtuple(
+    "_GraphParameters",
+    [
+        "G1",
+        "G2",
+        "G1_labels",
+        "G2_labels",
+        "nodes_of_G1Labels",
+        "nodes_of_G2Labels",
+        "G2_nodes_of_degree",
+    ],
+)
+_StateParameters = collections.namedtuple(
+    "_StateParameters",
+    ["mapping", "reverse_mapping", "T1", "T1_out", "T2", "T2_out"],
+)
+
+
+def vf2pp_mapping(G1, G2, node_labels=None, default_label=None):
+    """Return an isomorphic mapping between `G1` and `G2` if it exists.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism or monomorphism.
+
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
+
+    Returns
+    -------
+    Node mapping, if the two graphs are isomorphic. None otherwise.
+    """
+    try:
+        mapping = next(vf2pp_all_mappings(G1, G2, node_labels, default_label))
+        return mapping
+    except StopIteration:
+        return None
+
+
+def vf2pp_is_isomorphic(G1, G2, node_labels=None, default_label=None):
+    """Examines whether G1 and G2 are isomorphic.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism or monomorphism.
+
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
+
+    Returns
+    -------
+    True if the two graphs are isomorphic. False otherwise.
+    """
+    if vf2pp_mapping(G1, G2, node_labels, default_label) is not None:
+        return True
+    return False
+
+
+def vf2pp_all_mappings(G1, G2, node_labels=None, default_label=None):
+    """Yields all the possible mappings between G1 and G2.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism.
+
+    node_labels: string or None
+        The node attribute name within G that indicates node labels.
+        If None, then no node labels are used to compute isomorphisms.
+
+    default_label: string
+        The default label for nodes that have no label attribute value.
+    """
+    if G1.number_of_nodes() == 0 or G2.number_of_nodes() == 0:
+        return False
+
+    # Check that both graphs have the same number of nodes and degree sequence
+    if not _fast_precheck(G1, G2):
+        return False
```

Comment:
This `_fast_precheck` is now identical to the already existing `nx.faster_could_be_isomorphic` (which is in the module `isomorph.py` if you want to check it). So we should use that existing NX function.

oops -- we have to be careful here... the IND and MONO problems won't be able to use that function.  The `!=` becomes a `>=` in the degree check. Or maybe the whole test is no longer valid. Have you had to deal with this in your branch on IND? What's the right approach here?